### ISSUE TITLE
Image support in AI Asistent

### ISSUE THREAD

**Issue by KMJ-007** [zed-industries/zed#19810]
### Check for existing issues

- [X] Completed

### Describe the feature

# Feature Request: Image Pasting in Chat

Hey! 

While working with Zed's AI assistant for frontend tasks, I noticed that we can't paste images into the chat. It would be super helpful to share screenshots when discussing UI changes or bugs.

Currently, if I want to show the AI what I'm working on, I have to describe everything in text, which can be quite clumsy for visual content. 

## What I'd Love to See:

- **Paste screenshots directly into the chat (Cmd+V)**
- **Have the AI understand what's in the image**
- **Utilize images for frontend help, bug discussions, etc.**

Implementing this feature would make the AI assistant much more useful for frontend work!

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="646" alt="image" src="https://github.com/user-attachments/assets/6596aab1-7b89-4b50-83a6-b0257563b76d">

**Comment by cmomodo** [comment]
That would be very helpful

**Comment by SamuelLHuber** [comment]
would love to see this too in the AI chat. been using this a lot in all AI chats and would love it in Zed. Adding screenshots to get frontend work done is great

**Comment by Benjamin-van-Heerden** [comment]
Any updates on this?
For me it's not working for any provider (Linux)

**Comment by solanab** [comment]
I would love to have this feature!

**Comment by nicktanygin** [comment]
It would be great to paste images to Zed. E.g. for Expo apps in React Native it's very cumbersome to take a shortened error trace from iOS emulator. And traces in terminal often times have less context, so now I have to take a screenshot of the error trace in the emulator window, select text on image, and only then paste it to Zed.

**Comment by balboah** [comment]
I've been comparing different AI tools recently, and I noticed this is a great feature that works in claude code terminal but not in the AI panel. I prefer the more integrated Zed panel (and having a list of previous sessions) experience, but would probably stick to claude code until this and perhaps #32058 has been adressed

**Comment by kiahjh** [comment]
Just chiming in to agree with you all and say that this would be a super handy feature 👍

**Comment by notpeter** [comment]
Originally this was only support for Anthropic models, but now support has been added for other providers as well (Gemini, OpenAI, Mistral, Copilot Chat, et).

So I'm going to go ahead and close this as fixed/stale.
Thanks all!

**Comment by sudormdir** [comment]
> Originally this was only support for Anthropic models, but now support has been added for other providers as well (Gemini, OpenAI, Mistral, Copilot Chat, et).
> 
> So I'm going to go ahead and close this as fixed/stale. Thanks all!

How do I add images to ai chat?

**Comment by rgbkrk** [comment]
> How do I add images to ai chat?

Use paste. Drag and drop works too.

**Comment by mstarodub** [comment]
> Originally this was only support for Anthropic models, but now support has been added for other providers as well (Gemini, OpenAI, Mistral, Copilot Chat, et).
> 
> So I'm going to go ahead and close this as fixed/stale. Thanks all!

Doesn't seem to work for deepseek.

**Comment by niallobrien** [comment]
> > How do I add images to ai chat?
> 
> Use paste. Drag and drop works too.

Right, but you can't preview the actual image, or am I missing something?!
